crypto/asn1/asn1_lib.c

```diff
@@ -384,7 +388,8 @@ int ASN1_STRING_cmp(const ASN1_STRING *a, const ASN1_STRING *b)
 
     i = (a->length - b->length);
     if (i == 0) {
-        i = memcmp(a->data, b->data, a->length);
+        if (a->length != 0)
+            i = memcmp(a->data, b->data, a->length);
```

Comment:
I think this change was only neccessary when empty strings may be NULL.
Otherwise memcmp of 0 bytes is well-defined.

Reply:
IMO this cannot hurt.